### ISSUE TITLE
openssl vulnerabilities

### ISSUE THREAD

**Issue by romanb52** [confluentinc/librdkafka#4200]
Description
===========
librdkafka uses OpenSSL 3 prior to 3.0.8 which is vulnerable:
[CVE-2023-0286] - https://nvd.nist.gov/vuln/detail/CVE-2023-0286/
[CVE-2022-4450] - https://nvd.nist.gov/vuln/detail/CVE-2022-4450/
[CVE-2023-0215] - https://nvd.nist.gov/vuln/detail/CVE-2023-0215/

How to reproduce
================
No need, vulnerable libraries are part of librdkafka

Checklist
=========

Please provide the following information:

 - [x] librdkafka version (release number or git tag): 2.0.3
 - [x] Apache Kafka version: any
 - [x] librdkafka client configuration: any
 - [x] Operating system: any
 - [x] Provide logs (with `debug=..` as necessary) from librdkafka - not needed
 - [x] Provide broker log excerpts - not needed
 - [ ] Critical issue

**Comment by curtspiteri** [comment]
also librdkafka 2.0.2 uses libcurl version 7.86 which is also vulnerable as per https://curl.se/docs/vulnerabilities.html so it should be updated to the latest libcurl version.

**Comment by emasab** [comment]
@Vikash08Mishra  yes, we're scheduling the update for the release that's being published at the end of June. We try to update the precompiled binaries bundled in other Confluent clients as soon as possible. To meet the best security requirements, in terms of time to fix the vulnerability, it's also possible to install librdkafka linux packages and the latest versions of the dependencies. 

To use those:

**Python**
```
pip install --no-binary
```

**Go**
```
go build -tags dynamic
```

**.NET (in .csproj)**
```
    <PackageReference Include="Confluent.Kafka" Version="2.1.1" />
    <PackageReference Include="librdkafka.redist" Version="2.1.1" ExcludeAssets="All" />
```

**Comment by vdkranak** [comment]
Looks like Open SSL just released a new version 3.1.1 https://github.com/openssl/openssl/releases/tag/openssl-3.1.1

**Comment by romanb52** [comment]
Another vulnerability: [CVE-2023-2650](https://nvd.nist.gov/vuln/detail/CVE-2023-2650/)

**Comment by romanb52** [comment]
> @Vikash08Mishra yes, we're scheduling the update for the release that's being published at the end of June. We try to update the precompiled binaries bundled in other Confluent clients as soon as possible. To meet the best security requirements, in terms of time to fix the vulnerability, it's also possible to install librdkafka linux packages (from Confluent repositories) and the latest versions of the dependencies.
> 
> To use those:
> 
> **Python**
> 
> ```
> pip install --no-binary :all: confluent-kafka
> ```
> 
> **Go**
> 
> ```
> go build -tags dynamic
> ```
> 
> **.NET (in .csproj)**
> 
> ```
>     <PackageReference Include="Confluent.Kafka" Version="2.1.1" />
>     <PackageReference Include="librdkafka.redist" Version="2.1.1" ExcludeAssets="All" />
> ```

Any update please?

**Comment by romanb52** [comment]
Another one: CVE-2023-4807

**Comment by vivek-datadog** [comment]
Few more CVEs in openssl v3.0.8

- https://nvd.nist.gov/vuln/detail/CVE-2023-3817
- https://nvd.nist.gov/vuln/detail/CVE-2023-0464
- https://nvd.nist.gov/vuln/detail/CVE-2023-0465
- https://nvd.nist.gov/vuln/detail/CVE-2023-0466
- https://nvd.nist.gov/vuln/detail/CVE-2023-1255
- https://nvd.nist.gov/vuln/detail/CVE-2023-2650
- https://nvd.nist.gov/vuln/detail/CVE-2023-2975
- https://nvd.nist.gov/vuln/detail/CVE-2023-4807
- https://nvd.nist.gov/vuln/detail/CVE-2023-5363
- https://nvd.nist.gov/vuln/detail/CVE-2023-5678

**Comment by vivek-datadog** [comment]
> @Vikash08Mishra yes, we're scheduling the update for the release that's being published at the end of June. We try to update the precompiled binaries bundled in other Confluent clients as soon as possible. To meet the best security requirements, in terms of time to fix the vulnerability, it's also possible to install librdkafka linux packages (from Confluent repositories) and the latest versions of the dependencies.
> 
> To use those:
> 
> **Python**
> 
> ```
> pip install --no-binary :all: confluent-kafka
> ```
> 
> **Go**
> 
> ```
> go build -tags dynamic
> ```
> 
> **.NET (in .csproj)**
> 
> ```
>     <PackageReference Include="Confluent.Kafka" Version="2.1.1" />
>     <PackageReference Include="librdkafka.redist" Version="2.1.1" ExcludeAssets="All" />
> ```

Hello @emasab , reaching out to check on openssl version update timeline. Would this be taken care as part of https://github.com/confluentinc/librdkafka/pull/4303? I am particularly interested in the librdkafka with updated openssl for Windows environment.

**Comment by janjwerner-confluent** [comment]
Thank you for the report. We are in the process of resolving this issue.
Please see: https://github.com/confluentinc/librdkafka/pull/4706

**Comment by dpey2mtl** [comment]
Request to Upgrade OpenSSL to Latest Version

Hello Confluent Team,

I would like to request an upgrade of the OpenSSL package bundled with Confluent Kafka. Currently, version 3.0.8 is being used, which has known vulnerabilities that can pose security risks. Upgrading to version 3.0.13 or later would greatly enhance security.

Many users, including those utilizing the Datadog Agent, have flagged these vulnerabilities, and tools like Microsoft Defender have raised alerts regarding the presence of these outdated libraries.

I believe this upgrade is crucial for maintaining the security and integrity of applications relying on Confluent Kafka. 

Thank you for considering this request. I look forward to your response.

Best regards,
Didier

**Comment by janjwerner-confluent** [comment]
@dpey2mtl 
Please see https://github.com/confluentinc/librdkafka/issues/4786. The  discussion about openssl is continued there. 
cc @milindl @emasab
